include/Optimizer/Optimizer.hpp

```diff
@@ -1,5 +1,43 @@
+#ifndef GLU_OPTIMIZER_OPTIMIZER_HPP
+#define GLU_OPTIMIZER_OPTIMIZER_HPP
+
+#include <llvm/Support/Allocator.h>
+#include <llvm/Support/raw_ostream.h>
+
+namespace glu {
+
+class DiagnosticManager;
+class SourceManager;
+
+namespace gil {
+class Module;
+}
+
 class Optimizer {
```

Comment:
I'm not sure you need separate Optimizer and PassManager classes, you could merge this into the `PassManager` class